.github/workflows/push.yml

```diff
@@ -3,9 +3,16 @@ on: push
 name: Build and test
 jobs:
   buildDockerImage:
-    name: Build docker image
+    name: Publish master branch artifacts
     runs-on: ubuntu-latest
+    needs: [test_ui, test_go, test_go_more, test_go_oldest, test_windows, golangci, codeql, build_all]
+    if: github.event_name == 'push' && github.event.ref == 'refs/heads/master'
     steps:
-    - uses: actions/checkout@v4
-    - name: Build docker image
-      run: docker build -t yace --build-arg VERSION=${{github.event.release.tag_name}} .
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: prometheus/promci@52c7012f5f0070d7281b8db4a119e21341d43c91 # v0.4.5
+      - uses: ./.github/promci/actions/publish_master
+        with:
+          docker_hub_login: ${{ secrets.docker_hub_login }}
+          docker_hub_password: ${{ secrets.docker_hub_password }}
+          quay_io_login: ${{ secrets.quay_io_login }}
+          quay_io_password: ${{ secrets.quay_io_password }}
```

Comment:
Where are these secrets coming from? Are those already setup in this repo?

Reply:
They come from the prometheus community GitHub org vault.